### PR TITLE
[ENH] Lagllama interface for sktime 

### LINKED ISSUE #6142: [ENH] Interfacing LagLLama
Another foundation model we could interface is LagLLama: https://github.com/time-series-foundation-models/lag-llama

While the weights are hosted on HuggingFace, I suppose that they are not usable within the HFTransformersForecaster, since lagLlama is not implemented using the transformers library. But I suppose, It should not be that hard to implement it. A potential difficulty might be that this package is currently not released on pipit

### PR DESCRIPTION
**Pull request by pranavvp16**
#### Reference Issues/PRs

Closes #6142, closes #6659 

#### What does this implement/fix? Explain your changes.

This PR implements the LagLlama forecaster interface for sktime, providing a foundation model for univariate probabilistic time series forecasting
- Main forecaster class: `LagLlamaForecaster` in `sktime/forecasting/lagllama.py`
- Vendored library: Complete lag_llama library code in `sktime/libs/lag_llama/` (since lag-llama is not available as a PyPI package)

#### Does your contribution introduce a new dependency? If yes, which one?

Yes, this introduces soft dependencies (not core dependencies):
- `gluonts>=0.14.0` - Required for GluonTS dataset handling and transformations
- `torch` - Required for PyTorch model operations
- `lightning>=2.0,<2.6` - Required for PyTorch Lightning training framework
- `huggingface-hub` - Required for automatic checkpoint download from HuggingFace

#### What should a reviewer concentrate their feedback on?

1. Various data formats support, univariate, panel, and hierarchical 
2. Focus on fit_strategy ie. zero-shot and finetuning

#### Did you add any tests for the change?

The forecaster includes `get_test_params()` method with test parameter sets for:
- Zero-shot forecasting with default parameters
- Zero-shot with RoPE scaling enabled
- Fine-tuning with minimal epochs

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/sktime/sktime/blob/main/CONTRIBUTORS.md) with any new badges I've earned :-)
  How to: add yourself to the [all-contributors file](https://github.com/sktime/sktime/blob/main/.all-contributorsrc) in the `sktime` root directory (not the `CONTRIBUTORS.md`). Common badges: `code` - fixing a bug, or adding code logic. `doc` - writing or improving documentation or docstrings. `bug` - reporting or diagnosing a bug (get this plus `code` if you also fixed the bug in the PR).`maintenance` - CI, test framework, release.
  See here for [full badge reference](https://github.com/all-contributors/all-contributors/blob/master/docs/emoji-key.md)
- [x] Optionally, for added estimators: I've added myself and possibly to the `maintainers` tag - do this if you want to become the owner or maintainer of an estimator you added.
  See here for further details on the [algorithm maintainer role](https://www.sktime.net/en/latest/get_involved/governance.html#algorithm-maintainers).
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]. [BUG] - bugfix, [MNT] - CI, test framework, [ENH] - adding or improving code, [DOC] - writing or improving documentation or docstrings.

##### For new estimators
- [x] I've added the estimator to the API reference - in `docs/source/api_reference/taskname.rst`, follow the pattern.
- [x] I've added one or more illustrative usage examples to the docstring, in a pydocstyle compliant `Examples` section.
- [x] If the estimator relies on a soft dependency, I've set the `python_dependencies` tag and ensured
  dependency isolation, see the [estimator dependencies guide](https://www.sktime.net/en/latest/developer_guide/dependencies.html#adding-a-soft-dependency).

### CODE REVIEW

File: sktime/forecasting/lagllama.py [hunk 1]
```diff
@@ -0,0 +1,870 @@
...
+        "capability:pred_int:insample": True,
+        "authors": ["shlok191"],
+        "maintainers": ["shlok191"],
+        "python_version": None,
+        "python_dependencies": [
+            "gluonts>=0.14.0",
+            "torch",
+            "lightning>=2.0,<2.6",
+            "huggingface-hub",
+        ],
```
Comment (fkiraly) [inline sktime/forecasting/lagllama.py:257]: add a `tests:vm` tag so the testing runs on its own VM.

Comment (pranavvp16) [reply sktime/forecasting/lagllama.py:257]: ohk understood, read about the tag and made the change accordingly

**Comment by fkiraly** [comment]
is this still in progress, or have you abandoned it, @pranavvp16?

**Comment by pranavvp16** [comment]
> is this still in progress, or have you abandoned it, @pranavvp16?

I'm waiting for reviews on this, meanwhile I see the test's don't run on the CI even though I tested it on local

**Review by fkiraly — CHANGES_REQUESTED** [review]
Hm, something is not right with the tests. The test runs on python 3.13 and lower say that the estimator does not satisfy the deps, even though I just manually checked and it does. Is there a problem with `run_tests_vm` or `_check_estimator_deps`?

Also, we should try to remove `_BaseGlobalForecaster` inheritance and instead use the new API from https://github.com/sktime/sktime/pull/9056.

FYI @RecreationalMath, @SimonBlanke - can you make sense of why the tests do not seem to run properly? I am not able to get a diagnosis on this.

**Comment by pranavvp16** [comment]
> Hm, something is not right with the tests. The test runs on python 3.13 and lower say that the estimator does not satisfy the deps, even though I just manually checked and it does. Is there a problem with `run_tests_vm` or `_check_estimator_deps`?
> 
> Also, we should try to remove `_BaseGlobalForecaster` inheritance and instead use the new API from #9056.
> 
> FYI @RecreationalMath, @SimonBlanke - can you make sense of why the tests do not seem to run properly? I am not able to get a diagnosis on this.

After going through the latest design what I infer is, 
1. I need to update the _tags with pretrain related tag.
2. finetuning approach goes inside the _pretrain method i.e `_pretrain(self, y, X=None, fh=None) `
3. I'll remove the fine-tuning logic and the `fit_strategy` method from init which used to accept, zero-shot and finetuning
4. Inside fit we keep only zeroshot or use the predictor which got created if user called `pretrain`
5. Remove y from predict and predict_quantiles 
6. Inherit the BaseForecaster and remove the BaseGolbalForecaster

File: sktime/libs/lag_llama/data/augmentations/freq_mask.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    import torch
+else:
+    # Create dummy class when torch is not available
+    class torch:
+        @staticmethod
+        def no_grad():
+            def decorator(func):
+                return func
+
+            return decorator
```
Comment (jgyasu) [inline sktime/libs/lag_llama/data/augmentations/freq_mask.py]: I think it would be more elegant to use `_safe_import` here.

**Review by fkiraly — CHANGES_REQUESTED** [review]
IMPORTANT: the tests are not actually running on the CI. This indicates an import issue. Can you please check what is going on?

Other points:

* can you kindly check the license and add the lib correctly to the readme, and to the module `__init__`? Is this a permissive license?
* can you add the multiton pattern to ensure the model is not loaded multiple times on hierarchical data?
* you can add your name to `authors`, and you should perhaps be the sole `maintainer`, since @shlok191 is no longer active.

**Comment by pranavvp16** [comment]
> IMPORTANT: the tests are not actually running on the CI. This indicates an import issue. Can you please check what is going on?
> 
> Other points:
> 
> * can you kindly check the license and add the lib correctly to the readme, and to the module `__init__`? Is this a permissive license?
> * can you add the multiton pattern to ensure the model is not loaded multiple times on hierarchical data?
> * you can add your name to `authors`, and you should perhaps be the sole `maintainer`, since @shlok191 is no longer active.

Done i have made the requested changes.
1. The CI was unable to pickup tests due to skbase checking `huggingface-hub` against `huggingface_hub`
2. Pretrained Lag-Llama checkpoints unpickle classes from gluonts.torch.modules.loss, which GluonTS dropped in 0.15+, so torch.load raised ModuleNotFoundError. I fixed this by adding these method to the vendored lib and importing them just before the pretrained weights are loaded.
